java/memory/memory-core/src/main/java/org/apache/arrow/memory/util/StackTrace.java

```diff
@@ -19,12 +19,15 @@
 
 import java.util.Arrays;
 
+import org.checkerframework.checker.nullness.qual.Nullable;
+
+
 /**
  * Convenient way of obtaining and manipulating stack traces for debugging.
  */
 public class StackTrace {
 
-  private final StackTraceElement[] stackTraceElements;
+  private final @Nullable StackTraceElement [] stackTraceElements;
```

Comment:
This is the error without any annotation: found = what 3rd party library are offering in theirs API, required = what our apache arrow java library are offering
```
[ERROR]   found   : @Initialized @Nullable StackTraceElement @Initialized @NonNull []
[ERROR]   required: @Initialized @NonNull StackTraceElement @Initialized @NonNull []
```

Without any annotation, the framework assume that this variable is @ NonNull , in this case the Checker framework scanning `found` that `stackTraceElements ` will be nullable, for that reason I added @ Nullable annotations instead of @ NonNull default annotation.

Reply:
This part of the code `stackTraceElements = Arrays.copyOfRange(stack, 2, stack.length);`